v3/src/components/graph/components/graph-component.tsx

```diff
@@ -58,10 +63,10 @@ export const GraphComponent = observer(({broker, v2Document}: IProps) => {
       <InstanceIdContext.Provider value={instanceId}>
         <GraphLayoutContext.Provider value={layout}>
           <Graph model={defaultGraphModel}
-                 graphController={graphController}
-                 graphRef={graphRef}
-                 enableAnimation={enableAnimation}
-                 dotsRef={dotsRef}
+            graphController={graphController.current || getNewGraphController()}
```

Comment:
```suggestion
            graphController={graphController.current}
```
Rather than having two different places to create the graph controller, modify the `Graph` component to handle an `undefined` graph controller.